core/dbt/config/runtime.py

```diff
@@ -196,39 +215,18 @@ def validate(self):
         except ValidationError as e:
             raise DbtProjectError(validator_error_message(e)) from e
 
-    @classmethod
-    def _get_rendered_profile(
-        cls,
-        args: Any,
-        profile_renderer: ProfileRenderer,
-        profile_name: Optional[str],
-    ) -> Profile:
-
-        return Profile.render_from_args(args, profile_renderer, profile_name)
-
-    @classmethod
-    def get_profile(
-        cls: Type["RuntimeConfig"], args: Any, cli_vars: Dict[str, Any], raw_profile_name: str
-    ) -> Profile:
-        # build the profile using the base renderer and the one fact we know
-        # Note: only the named profile section is rendered. The rest of the
-        # profile is ignored.
-        profile_renderer = ProfileRenderer(cli_vars)
-        profile_name = profile_renderer.render_value(raw_profile_name)
-        profile = cls._get_rendered_profile(args, profile_renderer, profile_name)
```

Comment:
This method is overwritten in UnsetProfileConfig to return an `UnsetProfile` for tasks that don't require a profile. Is there still a place for something like `UnsetProfile` in the new high-level API to avoid loading the profile for tasks that don't require it?

Reply:
Rewired things to make sure existing test case passes